### PR TITLE
fix atomic_rmw for low-precision data types

### PR DESCRIPTION
**Pull request by tanyokwok**

### CODE REVIEW

File: tao_compiler/mlir/disc/transforms/disc_unhandled_atomic_rmw_converter.cc [hunk 1]
```diff
@@ -44,9 +44,12 @@ struct UnhandledAtomicRMWConverter
 
   LogicalResult matchAndRewrite(memref::AtomicRMWOp op,
                                 PatternRewriter& rewriter) const override {
+    Type elemTy = op.getType();
```
Comment (wyzero) [inline tao_compiler/mlir/disc/transforms/disc_unhandled_atomic_rmw_converter.cc:47]: better to add a i8 sum e2e test like here: https://github.com/alibaba/BladeDISC/blob/main/tao_compiler/mlir/disc/tests/tensorflow_ops/sum.cc#L25

File: tao_compiler/mlir/disc/tests/tensorflow_ops/sum.cc [hunk 2]
```diff
@@ -33,6 +33,30 @@ TEST(TFSumOpTest, ColReduceFullyDynamicShape2DF32) {
...
 
+// dynamic shape 2D column reduction test case
+TEST(TFSumOpTest, ColReduceFullyDynamicShape2DF16) {
+  EXPECT_TRUE(feature_test_main(
+      /*mlir_file_path*/ c_ft_path + "sum_col_d_2d_f16.mlir",
+      /*backend_types*/
+      kSupportedBackendList,
+      /*num_inputs*/ 1,
+      /*num_outputs*/ 1,
+      /*input_descriptors*/ {"110x100xf16_X"},
```
Comment (wyzero) [inline tao_compiler/mlir/disc/tests/tensorflow_ops/sum.cc:44]: cpu backend support f16？

File: tao_compiler/mlir/disc/transforms/disc_unhandled_atomic_rmw_converter.cc (hunk 1 above)
Comment (tanyokwok) [reply tao_compiler/mlir/disc/transforms/disc_unhandled_atomic_rmw_converter.cc:47]: done

File: tao_compiler/mlir/disc/tests/tensorflow_ops/sum.cc (hunk 2 above)
Comment (tanyokwok) [reply tao_compiler/mlir/disc/tests/tensorflow_ops/sum.cc:44]: I have no idea whether the CPU backend supports fp16, but the UT passed.

**Review by wyzero — APPROVED** [review]
LGTM

**Pull request merged**